src/Forte.Optimizely.ContentUsage/Api/Features/ContentType/ContentTypeUsagesDB.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using EPiServer.Construction;
+using EPiServer.Core;
+using EPiServer.Data;
+using EPiServer.DataAbstraction;
+using EPiServer.DataAccess.Internal;
+using EPiServer.Framework.Blobs;
+using EPiServer.ServiceLocation;
+using EPiServer.Web;
+
+namespace Forte.Optimizely.ContentUsage.Api.Features.ContentType;
+
+[ServiceConfiguration(Lifecycle = ServiceInstanceScope.Singleton)]
+// ReSharper disable once InconsistentNaming
+public class ContentTypeUsagesDB : ContentDB
```

Comment:
I wonder if we can get rid of this inheritance (ContentDB is from `Internal` namespace which is not safe to rely on - Optimizely does not commit on keeping backwards compatibility in such classes) ? As far as I can see all we need is GetExecutor method. Can we just try to inject IDatabaseExecutor  through regular DI ? Maybe even a IAsyncDatabaseExecutor and make whole code async?

Reply:
We could change it to async. But Opti makes their queries synchronious for some reason.